core/key.go

```diff
@@ -129,6 +132,27 @@ type CollectionIndexKey struct {
 
 var _ Key = (*CollectionIndexKey)(nil)
 
+// SchemaHistoryKey holds the pathway through the schema version history for
+// any given schema.
+//
+// The key points to the schema version id of the next version of the schema.
+// If a SchemaHistoryKey does not exist for a given SchemaVersionID it means
+// that that SchemaVersionID is for the latest version.
```

Comment:
It could only ever be an invalid key if something else has gone very horribly wrong in Defra, in which case this comment will be the least of anyone's problems :)